### PR TITLE
Issue #1 Resolved : Completed usage instructions in Readme file

### PR DESCRIPTION
**Pull request by HimandriSharma**
# Description

### Motivation
devs should know how to use this sdk

### Details
Write a short README explaining the main functions with an example

About # ([issue 1](https://github.com/GoodDollar/PaymentSDK/issues/4))

# How Has This Been Tested?

Readme file on github can be directly read.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @sirpy of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,82 @@
...
+For example:
+A Seller like e-commerce for "brand" would like to be able to request payment from his customers at Gooddollar.
+
+### Details
+#### Seller API
+A typescript NPM module that works both on browser/node that implements the following methods
+
+##### Method 1
+```
+generatePaymentLink(PaymentDetails:string)
```
Comment (sirpy) [inline README.md:17]: give example how to sue

File: README.md [hunk 2]
```diff
@@ -1,2 +1,82 @@
...
+```
+
+Creates a payment link.
+Example:
+[http://wallet.gooddollar.org:3000/? code=eyJtIjoiM2RQQjRxUjNBZUoyY0tzalJnYWp3dE1hUms3TjI0aFVMZzYiLCJhIjoiM...]()
+the query param code is the base64 encrypted value returned from generateCode.
+
+The generated code will return an object with the paymenDetails params and the ones that we already have
+
+Example:
```
Comment (sirpy) [inline README.md:27]: this is not an example this is an explanation of how the input is encoded in the base64

File: README.md [hunk 3]
```diff
@@ -1,2 +1,82 @@
...
+
+Same as above but create a QR containing the payment link from the generatePaymentLink method described above.
+Returns Data URL link (in string format).
+
+##### Method 3
+```
+validateTx(TxData):boolean
+```
+
+Reads fuse blockchain for txhash and check it has a Transfer log with the amount + sha3(invoiceid/data) paid to recipient in event log data.
```
Comment (sirpy) [inline README.md:49]: explains with example, how would you verify if someone paid a previously generated payment link

**Review by sirpy — COMMENTED** [review]
why does it include other commits also?

File: README.md (hunk 3 above)
Comment (HimandriSharma) [reply README.md:84]: > 

Not sure about this.

Comment (HimandriSharma) [reply README.md:84]: This function only cross checks the payment details. 
What do you mean by "someone paid a previously generated payment link"?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,135 @@
...
+
+Example:
+```
+{
+  "m":"3dPB4qR3AeJ2cKsjRgajwtMaRk7N24hULg6",
+  "a":"22200",
+  "r":"afds",
+  "cat":"Digital Services"
+}
+```
```
Comment (sirpy) [inline README.md:52]: this should match the input example

File: README.md [hunk 5]
```diff
@@ -1,2 +1,135 @@
...
...GxUnFZPKVHGiMlWcqLxRcaIyVXyhMlVMKlPFFw9rrXXJw1prXfKw1lqX/PBRxYnKicqJym9SmSpOVE4qTlQmlaliqnijYlJ5Q+VEZar4ouJE5YuKSWWq+DepnKhMFScVk8pfelhrrUse1lrrkoe11rrkh49U3qiYVKaKL1TeqJhUpoqTiknljYoTlaniv6zipoqbKk4q3qiYVKaKSeWNii8qJpXf9LDWWpc8rLXWJQ9rrXXJD39MZao4UZkqvlCZKt6o+KLiROVEZaqYVKaKE5UvVE4qTlTeqPhC5Y2KSeUNlS9UvqiYKn7Tw1prXfKw1lqXPKy11iU/fFTxmypOVL5QuaniDZWp4g2VqWJSmSpOKt5Q+S9TmSp+U8WJyknFGypvqJxUfPGw1lqXPKy11iUPa611yQ8fqfylipOKSWWqOFGZVKaKSeUmld+k8obKVPGGyn+ZyhsVk8qkMlV8oTJVnFRMKlPFpHLTw1prXfKw1lqXPKy11iU/XFZxk8oXFScqJxVfVEwqU8WkMlW8oTJVnKicVNxUMalMFV+oTBWTyk0Vk8pNFTep/KaHtda65GGttS55WGutS374ZSpvVPybKiaVqeINlaliUpkq/k0qX1RMKpPKVPGXKiaVqWJSmVS+qJhUJpUvVKaKv/Sw1lqXPKy11iUPa611yQ//n1GZKk4qJpWp4iaVqWJSOamYVG6qOFE5qXhDZao4qZhUpoqp4jepTBVTxaRyUnGiMqlMFZPKVPHFw1prXfKw1lqXPKy11iU//I+r+ELlC5WpYlI5qZhUTipOVKaKE5Wp4kRlqvg3qZxUTCpTxaRyUvGFyknFpDKpvFExqfymh7XWuuRhrbUueVhrrUt++GUVf0llqpgqJpWp4jdVTCo3VUwqU8VU8UbFGxWTyk0Vk8pJxaRyUjGpnFRMKjdVTConKn/pYa21LnlYa61LHtZa6xL7Bx+o/KWKSWWqmFSmihOVqWJSOamYVP5SxRsqU8WkMlVMKl9UTCpfVEwqU8WkclJxojJVnKhMFScqb1S8oTJVfPGw1lqXPKy11iUPa611if2Dtda64GGttS55WGutSx7WWuuSh7XWuuRhrbUueVhrrUse1lrrkoe11rrkYa21LnlYa61LHtZa65KHtda65GGttS55WGutSx7WWuuS/wept4ajhqVPWAAAAABJRU5ErkJggg==
```
Comment (sirpy) [inline README.md:77]: show the output as the image itself....

File: README.md (hunk 3 above)
Comment (sirpy) [reply README.md:84]: if you created a payment link with details X,Y,Z
how would you verify that a specific TX (ie txhash) matches the payment link details, meaning someone successfully paid for a service/product

**Review by sirpy — CHANGES_REQUESTED** [review]
@TomasGonzalez 
please pitch in

File: README.md (hunk 3 above)
Comment (HimandriSharma) [reply README.md:85]: I am checking that by fetching details associated with TxHash from fuse chain.

Comment (HimandriSharma) [reply README.md:85]: You mean I should invalidate the payment link once the transaction is done on it? @sirpy

Comment (sirpy) [reply README.md:85]: @TomasGonzalez

**Pull request merged**